packages/server/src/leaderboard/types.ts

```diff
@@ -0,0 +1,49 @@
+import { Solve } from '../database/solves'
+import { User } from '../database/users'
+import { Challenge } from '../challenges/types'
+import { ServerConfig } from '../config/types'
+
+export type WorkerRequest = {
+  solves: Solve[],
+  users: Pick<User, 'id' | 'name' | 'division'>[],
+  graphUpdate: number,
+  challenges: Challenge[],
+  config: ServerConfig
+}
+
+export type UserInfo = Pick<User, 'id' | 'name' | 'division'> & {
+  score: number,
+  lastSolve: number,
+  solvedChallengeIds: Challenge['id'][]
+}
+
+export type ExternalUserInfo = Pick<UserInfo, 'id' | 'name' | 'score'>
```

Comment:
I think the internal type should be the one with a prefix, since only the external type is used outside of this "module," and it seems awkward to be referencing an ExternalFoo as a consumer of the module.